### ISSUE TITLE
Escaping the wildcard character '.'

### ISSUE THREAD

**Issue by omar-kallas** [mhulden/pyfoma#13]
I have been trying to escape '.' to match a string with an actual period instead of using it as a wild card character, but all escaping methods don't work. I've tried all the following:
`x = FST.re("A'.'")
x = FST.re("A\.")
x = FST.re("A\\.")
x = FST.re("A\\\.")
x = FST.re(r"A\.")
x = FST.re(r"A.")`
None of them does what I want them to do. Interestingly, escaping two periods is possible using the single quotations but not one. So this code works to match two periods `x = FST.re("A'..'")`

**Comment by dhdaines** [comment]
Yes, this definitely can't be done with the way pyfoma currently works - `.` has a special meaning which is hardcoded in `apply` and `compose` among others, which is equivalent to "rho-matchers" in other FST libraries.

A workaround would be, as you say, to replace `.` with `'..'` in the input and use that in the FST.  It seems the regex compiler doesn't allow you to put literal backslashes in multi-character symbols (this is probably a bug) so you can't use `'\.'` unfortunately.

**Comment by dhdaines** [comment]
> Yes, this definitely can't be done with the way pyfoma currently works - `.` has a special meaning which is hardcoded in `apply` and `compose` among others, which is equivalent to "rho-matchers" in other FST libraries.

Correction: no, it is not equivalent to rho-matchers at all.  What a rho-matcher does in OpenFST is the following:

- During composition of two FSTs A and B, if the output symbol of an arc leaving the current state under consideration in A fails to match the input symbols of any arc leaving the current state under consideration in B, *but* there is an arc with a *rho* symbol as its input, then that arc will match.

What `.` does in pyfoma is something else:

- Nothing special happens during composition, strictly speaking
- *before* composition (and various other operations) on two FSTs A and B, if their alphabets are not equal, for each of A and B, every arc containing a `.` label will be duplicated to create new arcs for all of the symbols not in this FSTs vocabulary but present in the other FST.  Then the alphabets will be updated to be equal.
- in `apply`, `.` matches anything not in the FST's alphabet.

This is not at all the same thing!  Basically *rho*-matchers operate at the *state* level whereas `.` in pyfoma operates at the *FST level.  This is tightly bound to the fact that the alphabet of an FST is dynamic in pyfoma, which it is not in OpenFST.

**Comment by mhulden** [comment]
Should be possible now with [this commit.](https://github.com/mhulden/pyfoma/commit/27645c9165f8d812f31f9ab1c4b7425747fd9ac0)
